packages/app-project/src/screens/ClassifyPage/components/WorkflowMenu/WorkflowMenu.js

```diff
@@ -0,0 +1,80 @@
+import { useRouter } from 'next/router'
+import { arrayOf, shape, string } from 'prop-types'
+import React, { useState } from 'react'
+import { Modal } from '@zooniverse/react-components'
+
+import WorkflowSelector from '@shared/components/WorkflowSelector'
+import SubjectSetPicker from '@shared/components/SubjectSetPicker'
+
+/**
+  A popup menu which allows you to choose a workflow and optional subject set from the classify page.
+*/
+export default function WorkflowMenu({
+  headingBackground = 'brand',
+  titleColor = 'neutral-6',
+  workflowFromUrl,
+  workflows
+}) {
+  const router = useRouter()
+  const { owner, project } = router?.query || {}
+  const [ activeWorkflow, setActiveWorkflow ] = useState(workflowFromUrl)
+
+  function onSelectWorkflow(event, workflow) {
+    if (workflow.grouped) {
+      event.preventDefault()
+      setActiveWorkflow(workflow)
+      return false
+    }
+    return true
+  }
+
+  function onClose() {
+    setActiveWorkflow(null)
+  }
+
+  return (
+    <Modal
+      active
+      headingBackground={headingBackground}
+      title={activeWorkflow ? (activeWorkflow.displayName || 'Choose a subject set') : 'Choose a workflow'}
+      titleColor={titleColor}
+    >
+      {!activeWorkflow ?
+        <WorkflowSelector
+          onSelect={onSelectWorkflow}
+          workflows={workflows}
+        /> :
+        <SubjectSetPicker
+          onClose={onClose}
+          owner={owner}
+          project={project}
+          workflow={activeWorkflow}
+        />
+      }
+    </Modal>
+  )
+}
+
+const workflowType = shape({
+  displayName: string,
+  id: string
+})
+
+WorkflowMenu.propTypes = {
+  /**
+    Background colour of the title bar.
+  */
+  headingBackground: string,
```

Comment:
Wait, we have auto-documentation? I approve of these style of comments!

Reply:
I think it's JSDoc style commenting (I'm not sure.) The [Docs page documentation](https://storybook.js.org/docs/react/writing-docs/docs-page) doesn't explicitly mention it, but I figured it out from the examples. I think the storybook maintainers assume that everyone comments their code like this, as a matter of course.